com.unity.netcode.gameobjects/Runtime/Transports/UTP/UnityTransport.cs

```diff
@@ -489,6 +469,178 @@ private void DisposeInternals()
             TransportDisposed?.Invoke(GetInstanceID());
         }
 
+        /// <summary>
+        /// Get the <see cref="NetworkSettings"/> that will be used to create the underlying
+        /// <see cref="NetworkDriver"/> based on the current configuration of the transport. This
+        /// method is meant to be used with custom <see cref="INetworkStreamDriverConstructor"/>
+        /// implementations, where it can be used to modify the default parameters instead of
+        /// attempting to recreate them from scratch, which could be error-prone.
+        /// </summary>
+        /// <remarks>
+        /// The returned object is allocated using the <see cref="Allocator.Temp"/> allocator.
+        /// Do not hold a reference to it for longer than the current frame.
+        /// </remarks>
+        /// <returns>The default network settings.</returns>
+        /// <exception cref="Exception">
+        /// If the transport has an invalid configuration that prevents creating appropriate
+        /// settings for the driver. For example, if encryption is enabled but secrets haven't
+        /// been set, or if Relay is selected but the Relay server data hasn't been set.
+        /// </exception>
+        public NetworkSettings GetDefaultNetworkSettings()
+        {
+            var settings = new NetworkSettings();
+
+            // Basic driver configuration settings.
+            settings.WithNetworkConfigParameters(
+                maxConnectAttempts: m_MaxConnectAttempts,
+                connectTimeoutMS: m_ConnectTimeoutMS,
+                disconnectTimeoutMS: m_DisconnectTimeoutMS,
+                sendQueueCapacity: m_MaxPacketQueueSize,
+                receiveQueueCapacity: m_MaxPacketQueueSize,
+                heartbeatTimeoutMS: m_HeartbeatTimeoutMS);
+
+            // Configure Relay if in use.
+            if (m_ProtocolType == ProtocolType.RelayUnityTransport)
+            {
+                if (m_RelayServerData.Equals(default(RelayServerData)))
+                {
+                    throw new Exception("You must call SetRelayServerData() before calling StartClient() or StartServer().");
+                }
+                else
+                {
+                    settings.WithRelayParameters(ref m_RelayServerData, m_HeartbeatTimeoutMS);
+                }
+            }
+
+#if UNITY_MP_TOOLS_NETSIM_IMPLEMENTATION_ENABLED
+            // Latency, jitter and packet loss will be set by the network simulator in the tools
+            // package. We just need to initialize the settings since otherwise these features will
+            // not be enabled at all in the driver.
+            settings.WithSimulatorStageParameters(
+                maxPacketCount: 300, // TODO Is there any way to compute a better value?
```

Comment:
Small question: We're trying to avoid adding new TODOs to the codebase without a ticket or a person attached to them. Is this JIRA ticket worthy? Or just something to consider?

Reply:
Ah, my bad. I see where this was moved from. Resolving this!

Reply:
You're right that this TODO was there before, but might as well not carry it around when refactoring, so I updated that line and added a comment explaining how I calculated the new value.